Description: Build libhtsengine as shared library
 The upstream version provides only static library and Open JTalk
 uses it. This patch changes to build a shared library.
Author: Koichi Akabe <vbkaisetsu@gmail.com>
Last-Update: 2013-01-16
Forwarded: http://sourceforge.net/mailarchive/forum.php?thread_name=20130111191400.2cabb929b72dee8f9f50d12d%40gmail.com&forum_name=hts-engine-users

Index: htsengine/configure.ac
===================================================================
--- htsengine.orig/configure.ac
+++ htsengine/configure.ac
@@ -8,9 +8,13 @@ AC_COPYRIGHT(Copyright 2001-2015 Nagoya
 AC_COPYRIGHT(Copyright 2001-2008 Tokyo Institute of Technology)
 AM_INIT_AUTOMAKE
 
+LTVERSION="1:0:0"
+AC_SUBST(LTVERSION)
+
 # Checks for C compiler
 AC_PROG_CC
 AM_PROG_CC_C_O
+AM_PROG_LIBTOOL
 AC_PROG_INSTALL
 AC_PROG_RANLIB
 AN_MAKEVAR([AR], [AC_PROG_AR])
Index: htsengine/lib/Makefile.am
===================================================================
--- htsengine.orig/lib/Makefile.am
+++ htsengine/lib/Makefile.am
@@ -3,9 +3,10 @@ EXTRA_DIST = Makefile.mak
 
 AM_CPPFLAGS = -I @top_srcdir@/include
 
-lib_LIBRARIES = libHTSEngine.a
+lib_LTLIBRARIES = libHTSEngine.la
 
-libHTSEngine_a_SOURCES = HTS_audio.c  HTS_engine.c  HTS_hidden.h  HTS_misc.c  \
+libHTSEngine_la_LDFLAGS = -no-undefined -version-info $(LTVERSION)
+libHTSEngine_la_SOURCES = HTS_audio.c  HTS_engine.c  HTS_hidden.h  HTS_misc.c  \
                          HTS_pstream.c  HTS_sstream.c  HTS_model.c  HTS_vocoder.c \
                          HTS_gstream.c HTS_label.c
 
Index: htsengine/bin/Makefile.am
===================================================================
--- htsengine.orig/bin/Makefile.am
+++ htsengine/bin/Makefile.am
@@ -7,7 +7,7 @@ bin_PROGRAMS = hts_engine
 
 hts_engine_SOURCES = hts_engine.c 
 
-hts_engine_LDADD = ../lib/libHTSEngine.a
+hts_engine_LDADD = ../lib/libHTSEngine.la
 
 DISTCLEANFILES = *.log *.out *~
 
